src/styles/calendar.scss

```diff
@@ -0,0 +1,49 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+  max-width: $day-size * 7 + 6 * $gap;
+
+  @each $day, $size in $days-of-week {
+    &--start-day-#{$day} {
+      .calendar__day:first-child {
+        margin-left: $size * $start-day-size;
+      }
+    }
+  }
+
+  @each $length in $month-length {
+    &--month-length-#{$length} {
+      .calendar__day:nth-child(n + #{$length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    position: relative;
+    width: $day-size;
+    height: $day-size;
+    box-sizing: border-box;
+    background-color: $main-bg-color;
+    border: 1px solid $main-black-color;
+    transition: transform, background-color 0.5s ease;
```

Comment:
I think this transition doesn't work for `::before`

Reply:
how should it work for the before element?